tests/xgboost/test_xgboost_model_export.py

```diff
@@ -212,6 +213,88 @@ def test_model_save_persists_requirements_in_mlflow_model_directory(
     _compare_conda_env_requirements(xgb_custom_env, saved_pip_req_path)
 
 
+@pytest.mark.large
+def test_save_model_with_pip_requirements(xgb_model, tmpdir):
+    # Path to a requirements file
+    tmpdir1 = tmpdir.join("1")
+    f = tmpdir.join("requirements.txt")
+    f.write("a")
+    mlflow.xgboost.save_model(xgb_model.model, tmpdir1.strpath, pip_requirements=f.strpath)
+    reqs = _read_lines(tmpdir1.join("requirements.txt").strpath)
+    assert reqs == ["mlflow", "a"]
+
+    # List of requirements
+    tmpdir2 = tmpdir.join("2")
+    mlflow.xgboost.save_model(
+        xgb_model.model, tmpdir2.strpath, pip_requirements=[f"-r {f.strpath}", "b"]
+    )
+    reqs = _read_lines(tmpdir2.join("requirements.txt").strpath)
```

Comment:
Nice! For each of these test cases, can we also add logic to confirm that the `pip` section of `conda.yaml` is populated with the expected dependencies?

Reply:
Got it, I'll update the tests to ensure that.